src/main/java/org/betonquest/betonquest/events/DropItemEvent.java

```diff
@@ -47,40 +48,43 @@ public DropItemEvent(final Instruction instruction) throws InstructionParseExcep
         Bukkit.getPluginManager().registerEvents(this, BetonQuest.getInstance());
 
         questItems = instruction.getItemList();
-        location = instruction.getLocation("location");
-        isPrivate = instruction.hasArgument("isPrivate");
-        isIndestructible = instruction.hasArgument("nodespawn");
 
-        if (isPrivate) {
+        final String location = instruction.getOptional("location");
+        if (location == null) {
+            throw new InstructionParseException("No drop location given!");
+        } else {
+            this.location = instruction.getLocation(location);
+        }
+
+        final String isProtected = instruction.getOptional("protected");
+        this.isProtected = isProtected != null && isProtected.equals("true");
+
+        final String isPrivate = instruction.getOptional("private");
+        this.isPrivate = isPrivate != null && isPrivate.equals("true");
+
+        if (this.isPrivate) {
             if (Bukkit.getPluginManager().getPlugin("ProtocolLib") == null) {
                 throw new InstructionParseException("You need to install ProtocolLib to use private item drops!");
             } else {
-                hider = new EntityHider(BetonQuest.getInstance(), EntityHider.Policy.BLACKLIST);
+                hider = new EntityHider(BetonQuest.getInstance(), EntityHider.Policy.WHITELIST);
```

Comment:
Thank you for this, I just look at EntityHider code and just knowing about whitelist policy :D